geode-core/src/main/java/org/apache/geode/internal/cache/InitialImageOperation.java

```diff
@@ -1885,10 +1886,8 @@ public boolean executeWith(Object entList, int b) {
           if (lostMemberVersionID == null) {
             lostMemberVersionID = lostMemberID;
           }
-          // check to see if the region in this cache needs to synchronize with others
-          // it is possible that the cache is recover/restart of a member and not
-          // scheduled to synchronize with others
-          synchronizeIfNotScheduled(targetRegion, lostMemberID, lostMemberVersionID);
+          waitInAnotherThreadToCheckIfSynchronizationScheduled(targetRegion, lostMemberID,
```

Comment:
I don't understand what this has to do with GEODE-6899

Reply:
This is an enhancement for eshu's previous fix in GEODE-6802. His previous fix is the potential trigger of this bug, i.e. when the newly started member is trigger to call syncWith(). Then make this newly started member contains the entry with version. After that, the retry arrived and create a new version (It should not).